cli/types/cypress-eventemitter.d.ts

```diff
@@ -1,7 +1,23 @@
 // Cypress, cy, Log inherits EventEmitter.
 type EventEmitter2 = import("eventemitter2").EventEmitter2
 
-interface EventEmitter extends EventEmitter2 {
+type event = import("eventemitter2").event
+type eventNS = import("eventemitter2").eventNS
+type ListenerFn = import("eventemitter2").ListenerFn
+type Listener = import("eventemitter2").Listener
+type OnOptions = import("eventemitter2").OnOptions
+
+interface CyActions extends Cypress.Actions {
+  (event: event | eventNS, listener: ListenerFn, options?: boolean|OnOptions): this|Listener
+}
+
+interface CyEventEmitter {
+  once: CyActions
+  on: CyActions
+  off: CyActions
+  emit: EventEmitter2['emit']
+  removeListener: EventEmitter2['removeListener']
+  removeAllListeners: EventEmitter2['removeAllListeners']
```

Comment:
Our docs say that Cypress/cy are "standard Node event emitters": https://docs.cypress.io/api/events/catalog-of-events#Binding-to-Events

I'm wondering if we need to either:
* Change the wording there if we only intend to support a subset of the API
* Add more functions here to align with what [Node's event emitter](https://nodejs.org/dist/latest-v16.x/docs/api/events.html) provides, like `addListener`. The other missing APIs are probably less frequently used, but it's hard to say.

Reply:
I decided to change the type to `Omit<EventEmitter2, 'waitFor'>`.

Because other functions are harmless, they're really unlikely to be used in the real world cases, though.